guardian/testapp/tests/test_shortcuts.py

```diff
@@ -124,6 +130,34 @@ def test_deprecation_warning(self):
             self.assertEqual(len(warns), 1)
             self.assertTrue(isinstance(warns[0].message, DeprecationWarning))
 
+    def test_assign_perm_from_user_group_object(self):
+        post = Post.objects.create(title='Rouge One')
+        user = User.objects.create(username='Jyn Erso')
+        group = Group.objects.create(name='Rebel Alliance')
+        user_group_object = UserGroupObject.objects.create(user=user, group=group, content_object=post)
+        assign_perm_from_user_group_object('testapp.add_post', user_group_object)
+        self.assertTrue(user.has_perm('testapp.add_post', post))
+        user_group_object.delete()
+        self.assertFalse(user.has_perm('testapp.add_post', post))
```

Comment:
since permissions are cached, I would expect it to fail unless you reload the user. Wondering why it works
https://docs.djangoproject.com/en/1.9/topics/auth/default/#permission-caching